bin/genome-env-next

```diff
@@ -429,43 +433,76 @@ function setup_env {
 
     # remove deployed bin directory from PATH
     PATH="$(echo $PATH | tr : "\n" | $GREP -v "$SNAPSHOT_BIN" | tr '\n' : | sed 's/:$//')"
-    PATH=$UR_REPO/bin:$PATH
-    PATH=$WF_REPO/bin:$PATH
-    PATH=$BASE_DIR/bin:$PATH
-    export PATH
+    export PATH=$BASE_DIR/bin:$PATH
 
     # remove deployed lib/perl directory from PATH
-    PERL5LIB="$(echo $PERL5LIB | tr : "\n" | $GREP -v "$SNAPSHOT_LIB" | tr '\n' : | sed 's/:$//')"
+    export PERL5LIB="$(echo $PERL5LIB | tr : "\n" | $GREP -v "$SNAPSHOT_LIB" | tr '\n' : | sed 's/:$//')"
 
-    for MODULE in UR Workflow Genome
-    do
-        if genome-perl -M$MODULE -e 1 2> /dev/null
+    if genome-perl -MGenome -e 1 2> /dev/null
+    then
+        fatal $BUILD_ERR "still found Genome after removing deployed snapshot" $(genome-perl -S wtf Genome)
+    fi
+
+    if test -z "$DISABLE_UR"
+    then
+        export PATH=$UR_REPO/bin:$PATH
+        if genome-perl -MUR -e 1 2> /dev/null
         then
-            fatal $BUILD_ERR "still found $MODULE after removing deployed snapshot"
+            fatal $BUILD_ERR "still found UR after removing deployed snapshot"
         fi
-    done
+    fi
 
-    PERL5LIB=$UR_REPO/lib:$PERL5LIB
-    PERL5LIB=$BASE_DIR/configuration-manager/lib:$PERL5LIB
-    PERL5LIB=$WF_REPO/lib:$PERL5LIB
-    PERL5LIB=$BASE_DIR/lib/perl:$PERL5LIB
-    export PERL5LIB
+    if test -z "$DISABLE_WF"
+    then
+        export PATH=$WF_REPO/bin:$PATH
+        if genome-perl -MWorkflow -e 1 2> /dev/null
+        then
+            fatal $BUILD_ERR "still found Workflow after removing deployed snapshot"
+        fi
+    fi
 
-    for MODULE in UR Workflow Genome
-    do
-        if ! genome-perl -M$MODULE -e 1 2> /dev/null
+    if which genome | $GREP -q "$SNAPSHOT_BIN"
+    then
+        fatal $BUILD_ERR "genome found in $SNAPSHOT_BIN"
+    fi
+
+    if test -z "$DISABLE_UR"
+    then
+        export PERL5LIB=$UR_REPO/lib:$PERL5LIB
+
+        if ! genome-perl -MUR -e 1 2> /dev/null
         then
-            fatal $BUILD_ERR "failed to load $MODULE"
+            genome-perl -MUR -e 1
```

Comment:
What is this second `genome-perl` without the `2> /dev/null` for?

In general, would it be worthwhile to extract `genome-perl -M$MODULE -e 1 2> /dev/null` into a function?